### ISSUE TITLE
[2.1.2] template and copy still uses old CWD 

### ISSUE THREAD

**Issue by resmo** [ansible/ansible#17896]
##### ISSUE TYPE

- Bug Report
##### ANSIBLE VERSION

```
2.1.2 
```
##### CONFIGURATION

ansible.cfg with an special config for a basion jump host.

``` ini
[ssh_connection]
ssh_args = -F ssh.config
```

playbooks and ssh.config are located in a subdir `playbooks/`
##### OS / ENVIRONMENT

N/A
##### SUMMARY

With e2f17f8d9b212b1e8d65c12a778d036753831ab6 the CWD changed to the location where the playbooks are located, in this case ./playbooks. ssh.config has been moved there and this worked. 

However for the plugins setup, copy and template, ansible still expects the ssh.config to be located in the location as in 2.1.1. 
##### STEPS TO REPRODUCE

```
echo www.example.com > hosts
mkdir -p playbooks/files
echo "foo" > playbooks/files/test.txt
touch playbooks/ssh.config # can be empty!

cat ansible.cfg
[ssh_connection]
ssh_args = -F ssh.config 

cat playbooks/test.yml 

---
- hosts: all
  tasks:
  - copy: src=files/test.txt dest=/tmp

ansible-playbook playbooks/test.yml -i hosts --user root
```

after cp the ssh.config to the old "ansible 2.0" location, it works again!

```
cp playbooks/ssh.config .   
```

##### EXPECTED RESULTS

```
PLAY [all] *********************************************************************

TASK [setup] *******************************************************************
ok: [build-01.stxt.media.int]

TASK [copy] ********************************************************************
changed: [build-01.stxt.media.int]

PLAY RECAP *********************************************************************
build-01.stxt.media.int    : ok=2    changed=1    unreachable=0    failed=0  
```
##### ACTUAL RESULTS

```
 $ ansible-playbook playbooks/test.yml -i hosts --user root

PLAY [all] *********************************************************************

TASK [setup] *******************************************************************
fatal: [build-01.stxt.media.int]: UNREACHABLE! => {"changed": false, "msg": "Failed to connect to the host via ssh.", "unreachable": true}
    to retry, use: --limit @/home/resmo/Projects/swisstxt/tmp/playbooks/test.retry

PLAY RECAP *********************************************************************
build-01.stxt.media.int    : ok=0    changed=0    unreachable=1    failed=0   

```

With `gather_facts: no`:

```
PLAY [all] *********************************************************************

TASK [copy] ********************************************************************
fatal: [build-01.stxt.media.int]: UNREACHABLE! => {"changed": false, "msg": "SSH Error: data could not be sent to the remote host. Make sure this host can be reached over ssh", "unreachable": true}
    to retry, use: --limit @/home/resmo/Projects/swisstxt/tmp/playbooks/test.retry

PLAY RECAP *********************************************************************
build-01.stxt.media.int    : ok=0    changed=0    unreachable=1    failed=0
```

**Comment by bcoca** [comment]
dupe of #17770

**Comment by resmo** [comment]
as the bug still exists with commit wihich resolves #17770 

```
 $ ansible --version
ansible 2.1.2.0 (stable-2.1 2a234c1ff9) last updated 2016/10/04 16:41:45 (GMT +200)
  lib/ansible/modules/core: (detached HEAD 2b9300ec28) last updated 2016/09/19 14:40:47 (GMT +200)
  lib/ansible/modules/extras: (detached HEAD 35210ea450) last updated 2016/09/19 14:40:47 (GMT +200)
  config file = /home/resmo/Projects/swisstxt/tmp/ansible.cfg
  configured module search path = Default w/o overrides
 resmo@stxtc00039  ~/Projects/swisstxt/tmp  
 $ ansible-playbook playbooks/test.yml -i hosts --user root      

PLAY [all] *********************************************************************

TASK [setup] *******************************************************************
fatal: [build-01.stxt.media.int]: UNREACHABLE! => {"changed": false, "msg": "Failed to connect to the host via ssh.", "unreachable": true}
    to retry, use: --limit @/home/resmo/Projects/swisstxt/tmp/playbooks/test.retry

PLAY RECAP *********************************************************************
build-01.stxt.media.int    : ok=0    changed=0    unreachable=1    failed=0   

 $ cp playbooks/ssh.config .
 $ ansible-playbook playbooks/test.yml -i hosts --user root

PLAY [all] *********************************************************************

TASK [setup] *******************************************************************
ok: [build-01.stxt.media.int]

TASK [copy] ********************************************************************
ok: [build-01.stxt.media.int]

PLAY RECAP *********************************************************************
build-01.stxt.media.int    : ok=2    changed=0    unreachable=0    failed=0   
```

**Comment by resmo** [comment]
I created an sample repo for this issue.
